### PR TITLE
New type to ensure a file is downloaded

### PR DESCRIPTION
**Pull request by frdmn**
Including an Content-Length check to ensure the local and remote files
are the same.

### CODE REVIEW

File: types/download.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    if bake [ -f \"$target\" ]; then
+      existing_contentlength=$(ls -al "$target" | tr -s ' ' | cut -d' ' -f5)
+
+      remote_contentlength=$(curl -sI "$url" | grep Content-Length | tr -s ' ' | cut -d' ' -f2)
+      remote_contentlength=${remote_contentlength%%[^0-9]*} # remove all non-numerical characters
+
+      if [ "$existing_contentlength" != "$remote_contentlength" ]; then
+        echo "received Content-Length for existing file: $existing_contentlength bytes"
+        echo "expected Content-Length from remote: $remote_contentlength bytes"
+        return $STATUS_MISSING
```
Comment (mattly) [inline types/download.sh]: This is really more of a CONFLICT_UPGRADE. My plans for bork eventually include flags for the runner to automatically resolve mismatch/conflict but I haven't gotten there yet.

**Comment by mattly** [comment]
I'd like to get some tests around this before merging, and i'm not really up for doing that tonight. 

One of my "really I'd love to do this before some kind of official 'release'" goals was to encapsulate the pattern of download, unarchive, configure, make, install (see https://github.com/mattly/dotfiles-email/blob/master/bork_setup.sh#L3, worked with an older bork) but this is a really good stepping stone.

Another thing I want to do with http downloads is provide a fallback to wget if curl is unavailable. I was thinking of a wrapper akin to the md5cmd or permissions helper.

Things with work should clear up in a few weeks, and I can try for doing the tests then to get this merged.

**Comment by frdmn** [comment]
@mattly no hurry buddy. Take your time.

**Comment by frdmn** [comment]
Any news on this one? Would love to trash my fork :D

**Comment by mattly** [comment]
replaced with #51

**Pull request closed without merging**